.travis.yml

```diff
@@ -18,9 +18,7 @@ install:
   - molecule --version
 
 script:
-  - molecule lint -s openstack
```

Comment:
Okay, this is making me think that there are several molecule issues that may need to be looked at.

```
--> Scenario: 'default'
--> Action: 'lint'
--> Executing Yamllint on files found in /oasis/roles/testrole/...
Lint completed successfully.
--> Executing Flake8 on files found in /oasis/roles/testrole/molecule/default/tests/...
Lint completed successfully.
--> Executing Ansible Lint on /oasis/roles/testrole/molecule/default/playbook.yml...
Lint completed successfully.
```

- yamllint runs on the entire role, which is the behavior that prompted me to make this change.
  It probably should behave as you've described. It is run using the lint config defined in the targeted scenario, though, so you can get potentially inconsistent results if different scenarios have different yamllint configs. This skeleton thankfully guards against that.
- flake8 only gets run if testinfra discovers tests in a scenario's `tests` dir (an empty `scenario/tests/test_nothing.py` works for this), there's no way to run it that I can see without a verifier like testinfra running first
- ansible-lint only lints the playbook.yml. I made a prepare.yml and a playbook.yml with the same lint errors, but only the playbook error was reported. Granted, I haven't take a real close look, but this appears to be confirmed in [the code](https://github.com/ansible/molecule/blob/master/molecule/provisioner/lint/ansible_lint.py#L135-L146)

Based on that, it's probably worth adding a comment to the travis config that it's a good idea to lint every scenario. After that, a stroll through the molecule issue tracker is probably in order.

Reply:
Huh, I put the flake8 call in *just in case* Molecule doesn't flake libraries, plugins, filters, etc. I had no reason to suspect that it didn't, except for my overall suspicions about the reliability of other developers' code. Before, after, I'm not too concerned with it. I do think it makes sense for lint/static analysis to be done before active testing, though, so I'm down with that.